adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m

```diff
@@ -63,22 +106,56 @@ + (GADVersionNumber)adapterVersion {
 
 - (void)loadBannerForAdConfiguration:(GADMediationBannerAdConfiguration *)adConfiguration
                    completionHandler:(GADMediationBannerLoadCompletionHandler)completionHandler {
-  // TODO: Start loading a bidding banner ad.
+    [GADMediationAdapterPangle setCOPPA:adConfiguration.childDirectedTreatment.integerValue];
+    _bannerRenderer = [[GADPangleRTBBannerRenderer alloc] init];
+    [_bannerRenderer renderBannerForAdConfiguration:adConfiguration completionHandler:completionHandler];
 }
 
 - (void)loadInterstitialForAdConfiguration:(GADMediationInterstitialAdConfiguration *)adConfiguration
                          completionHandler:(GADMediationInterstitialLoadCompletionHandler)completionHandler {
-  // TODO: Start loading a bidding interstitial ad.
-}
-
-- (void)loadNativeAdForAdConfiguration:(GADMediationNativeAdConfiguration *)adConfiguration
-                     completionHandler:(GADMediationNativeLoadCompletionHandler)completionHandler {
-  // TODO: Start loading a bidding native ad.
+    [GADMediationAdapterPangle setCOPPA:adConfiguration.childDirectedTreatment.integerValue];
+    _interstitialRenderer = [[GADPangleRTBInterstitialRenderer alloc] init];
+    [_interstitialRenderer renderInterstitialForAdConfiguration:adConfiguration completionHandler:completionHandler];
 }
 
 - (void)loadRewardedAdForAdConfiguration:(GADMediationRewardedAdConfiguration *)adConfiguration
                        completionHandler:(GADMediationRewardedLoadCompletionHandler)completionHandler {
-  // TODO: Start loading a bidding rewarded ad.
+    [GADMediationAdapterPangle setCOPPA:adConfiguration.childDirectedTreatment.integerValue];
+    _rewardedRenderer = [[GADPangleRTBRewardedRenderer alloc] init];
+    [_rewardedRenderer renderRewardedAdForAdConfiguration:adConfiguration completionHandler:completionHandler];
+}
+
++ (void)setCOPPA:(NSInteger)COPPA {
+    if (COPPA != 0 && COPPA != 1 && COPPA != -1) {
+        PangleLog(@"Invalid COPPA value. Pangle SDK only accepts -1, 0 or 1.");
+        return;
+    }
+    if (BUAdSDKManager.initializationState == BUAdSDKInitializationStateReady) {
+        [BUAdSDKManager setCoppa:COPPA];
+    }
+    _coppa = COPPA;
+}
```

Comment:
This will actually introduce a bug since if `adConfiguration.childDirectedTreatment` is `nil`, then `adConfiguration.childDirectedTreatment.integerValue` will be `0`, which is the same as  `adConfiguration.childDirectedTreatment` is `NO` (i.e. child direct treatment is not enabled).

I think we should update this method to a private method that takes in a `NSNumber`, and have the below implementation:
```
- (void)setCoppa:(NSNumber)coppa {
    if (coppa == nil) {
        _coppa = -1;
    } else if (coppa.integerValue == 1) {
       _coppa = 1;
    } else if (coppa.integerValue == 0) {
       _coppa = 0;
    } else {
       PangleLog(@"Invalid COPPA value. Pangle SDK only accepts -1, 0 or 1.");
       return;
    }
    if (BUAdSDKManager.initializationState == BUAdSDKInitializationStateReady) {
        [BUAdSDKManager setCoppa:COPPA];
    }
}
```

Reply:
I update this invoke way ：
`
 if (adConfiguration.childDirectedTreatment) {
        [GADMediationAdapterPangle setCOPPA:adConfiguration.childDirectedTreatment.integerValue];
    }      
`